src/app/password-token/password-token.service.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable } from "@nestjs/common";
+import { PrismaService } from "nestjs-prisma";
+import { PasswordToken } from "@prisma/client";
+import * as crypto from "crypto";
+import * as dayjs from "dayjs";
+
+@Injectable()
+export class PasswordTokenService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  private generateToken() {
+    return crypto.randomBytes(64).toString("hex");
+  }
+
+  // ttl in seconds
+  private hasExired(ttl: number, token: PasswordToken) {
+    if (!token) return true;
+    return dayjs(token.createdAt).add(ttl, "s").isBefore(dayjs());
+  }
+
+  async create(forUserId: string) {
+    await this.prisma.passwordToken.deleteMany({
+      where: {
+        userId: forUserId
+      }
+    });
+    const passwordToken = await this.prisma.passwordToken.create({
+      data: {
+        token: this.generateToken(),
+        userId: forUserId
+      }
+    });
+    return passwordToken.token;
+  }
+
+  async verifyAndRevoke(userId: string, token: string) {
+    const realToken = await this.prisma.passwordToken.findUnique({
+      where: {
+        userId
+      }
+    });
+
+    if (!realToken) return false;
```

Comment:
I'm getting a lil confused in these if checks.. what is the flow of the logic ( for my understanding)?

Reply:
This fn receives token provided by the user and that user's id. It then fetches that user's real token from database. Then some checks are done:
1) If  real(database) token does not exist, reject.
2) If real token is not same as user's token, reject.
3) If real token has expired, reject.

Else verify token as valid and remove the token from database.

Reply:
there are some things that we discussed that i need to do like jwt token and stuff, and im thinking it will be better if i do that in a seperate pr